### PR TITLE
refactor: change `check` return type

### PR DESCRIPTION
**Pull request by jaschdoc**
Closes https://github.com/flix/flix/issues/9141

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/api/Bootstrap.scala [hunk 1]
```diff
@@ -681,14 +683,12 @@ class Bootstrap(val projectPath: Path, apiKey: Option[String]) {
...
-    Validation.mapN(flix.check()) {
-      root =>
-        HtmlDocumentor.run(root, packageModules)(flix)
-    }.toHardResult match {
-      case Result.Ok(_) =>
-        Validation.success(())
-      case Result.Err(errors) =>
-        Validation.toHardFailure(BootstrapError.GeneralError(flix.mkMessages(errors)))
+    val (result, errors) = flix.check()
+    result.foreach(HtmlDocumentor.run(_, packageModules)(flix))
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/Bootstrap.scala:687]: We should not run the HtmlDocumenter if there are errors.

File: main/src/ca/uwaterloo/flix/api/Flix.scala [hunk 2]
```diff
@@ -558,7 +559,11 @@ class Flix {
     }
 
     // Return the result (which could contain soft failures).
-    result
+    val errors = result.errors.toList
+    result match {
+      case Validation.HardFailure(_) => (None, errors)
+      case _ => (Some(result.unsafeGet), errors)
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/Flix.scala:565]: Make Success public and match on it

**Comment by magnus-madsen** [comment]
@jaschdoc Test failures

File: main/src/ca/uwaterloo/flix/api/Bootstrap.scala (hunk 1 above)
Comment (jaschdoc) [reply main/src/ca/uwaterloo/flix/api/Bootstrap.scala:687]: Ok but it did that before :)

**Comment by jaschdoc** [comment]
@magnus-madsen only expected failures

File: main/src/ca/uwaterloo/flix/api/Bootstrap.scala (hunk 1 above)
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/api/Bootstrap.scala]: Really? Thats crazy.

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**